(window.webpackJsonp=window.webpackJsonp||[]).push([[9],{366:function(e,t,a){"use strict";a.r(t);var s=a(45),r=Object(s.a)({},(function(){var e=this,t=e.$createElement,a=e._self._c||t;return a("ContentSlotsDistributor",{attrs:{"slot-key":e.$parent.slotKey}},[a("h1",{attrs:{id:"general-usage"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#general-usage"}},[e._v("#")]),e._v(" General usage")]),e._v(" "),a("p",[a("a",{attrs:{href:"https://github.com/davidebolo1993/TREADMILL",target:"_blank",rel:"noopener noreferrer"}},[e._v("TREADMILL"),a("OutboundLink")],1),e._v(" is built on 3 modules:")]),e._v(" "),a("ol",[a("li",[a("a",{attrs:{href:"https://github.com/davidebolo1993/TREADMILL/tree/master/TREADMILL/BASIC",target:"_blank",rel:"noopener noreferrer"}},[e._v("BASIC"),a("OutboundLink")],1),e._v(", the 'BAm StatIstiCs' module")]),e._v(" "),a("li",[a("a",{attrs:{href:"https://github.com/davidebolo1993/TREADMILL/tree/master/TREADMILL/RACE",target:"_blank",rel:"noopener noreferrer"}},[e._v("RACE"),a("OutboundLink")],1),e._v(", the 'ReAds ClusterEr' module")]),e._v(" "),a("li",[a("a",{attrs:{href:"https://github.com/davidebolo1993/TREADMILL/tree/master/TREADMILL/TRAP",target:"_blank",rel:"noopener noreferrer"}},[e._v("TRAP"),a("OutboundLink")],1),e._v(", the 'Tandem RepeAts Profiler' module")])]),e._v(" "),a("p",[e._v("It then wraps "),a("a",{attrs:{href:"https://github.com/bioinfomaticsCSU/deepsignal",target:"_blank",rel:"noopener noreferrer"}},[e._v("DeepSignal"),a("OutboundLink")],1),e._v(" to perform allele-specific methylation profiling of CG motifs")]),e._v(" "),a("h2",{attrs:{id:"treadmill-basic"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#treadmill-basic"}},[e._v("#")]),e._v(" TREADMILL BASIC")]),e._v(" "),a("p",[e._v("BASIC allows users to retrieve essential statistics from targeted ONT sequencing runs in "),a("a",{attrs:{href:"https://samtools.github.io/hts-specs/SAMv1.pdf",target:"_blank",rel:"noopener noreferrer"}},[e._v("BAM format"),a("OutboundLink")],1),e._v(" once a proper "),a("a",{attrs:{href:"https://www.ensembl.org/info/website/upload/bed.html",target:"_blank",rel:"noopener noreferrer"}},[e._v("BED file"),a("OutboundLink")],1),e._v(" specifying repeat coordinates in enriced targets is provided. The input BAM file must be sorted and MD/NM-tagged.")]),e._v(" "),a("div",{staticClass:"language-bash extra-class"},[a("pre",{pre:!0,attrs:{class:"language-bash"}},[a("code",[a("span",{pre:!0,attrs:{class:"token comment"}},[e._v("#TREADMILL BASIC --help")]),e._v("\n\nTREADMILL BASIC -bam "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("MD.NM.TAGGED.BAM"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -bed "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("TARGETS.BED"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -o "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("OUTPUTBASIC.JSON.GZ"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v("\n")])])]),a("p",[e._v("BASIC dumps computed statistics to a (gzipped) "),a("a",{attrs:{href:"https://www.json.org/json-en.html",target:"_blank",rel:"noopener noreferrer"}},[e._v("JSON file"),a("OutboundLink")],1),e._v(". This JSON file contains:")]),e._v(" "),a("ol",[a("li",[e._v("The number of unmapped, primary, secondary and supplementary reads in the BAM file")]),e._v(" "),a("li",[e._v("The number of on-target and off-target primary alignments")]),e._v(" "),a("li",[e._v("For on-target reads, the number of matches, mismatches, insertions, deletions and soft-clipped bases")]),e._v(" "),a("li",[e._v("The lenght, quality and percent identity of on-target reads together with their per-base coverage")])]),e._v(" "),a("p",[e._v("These informations can be summarized, for one ore more JSON file, using the companion "),a("a",{attrs:{href:"https://github.com/davidebolo1993/TREADMILL/blob/master/scripts/plotstats.R",target:"_blank",rel:"noopener noreferrer"}},[e._v("plotstats R script"),a("OutboundLink")],1),e._v(" provided in the "),a("a",{attrs:{href:"https://github.com/davidebolo1993/TREADMILL/tree/master/scripts",target:"_blank",rel:"noopener noreferrer"}},[e._v("scripts folder of TREADMILL"),a("OutboundLink")],1),e._v(".")]),e._v(" "),a("div",{staticClass:"language-bash extra-class"},[a("pre",{pre:!0,attrs:{class:"language-bash"}},[a("code",[a("span",{pre:!0,attrs:{class:"token comment"}},[e._v("#Rscript TREADMILL/scripts/plotstats.R -h")]),e._v("\n\nRscript TREADMILL/scripts/plotstats.R -j "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("BASIC.JSON1.GZ,"),a("span",{pre:!0,attrs:{class:"token punctuation"}},[e._v("..")]),e._v("."),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -l "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("EXP1,"),a("span",{pre:!0,attrs:{class:"token punctuation"}},[e._v("..")]),e._v("."),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -o "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("BASIC.STATS.PDF"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v("\n")])])]),a("p",[e._v("Use cases for BASIC are described in the "),a("RouterLink",{attrs:{to:"/usecases/usecases.html"}},[e._v("Use cases section")]),e._v(".")],1),e._v(" "),a("h2",{attrs:{id:"treadmill-race"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#treadmill-race"}},[e._v("#")]),e._v(" TREADMILL RACE")]),e._v(" "),a("p",[e._v("RACE is the core module of TREADMILL. For each target in BED, TREADMILL RACE first applies a decoy-aligment strategy, following the principles described in "),a("a",{attrs:{href:"https://genomebiology.biomedcentral.com/articles/10.1186/s13059-018-1505-2",target:"_blank",rel:"noopener noreferrer"}},[e._v("STRetch"),a("OutboundLink")],1),e._v(". It then runs a clustering algorithm - by default, the "),a("a",{attrs:{href:"https://en.wikipedia.org/wiki/DBSCAN",target:"_blank",rel:"noopener noreferrer"}},[e._v("Density-Based Spatial Clustering of Applications with Noise"),a("OutboundLink")],1),e._v(" - to group reads with similar repeat content.")]),e._v(" "),a("div",{staticClass:"language-bash extra-class"},[a("pre",{pre:!0,attrs:{class:"language-bash"}},[a("code",[a("span",{pre:!0,attrs:{class:"token comment"}},[e._v("#TREADMILL RACE --help")]),e._v("\n\nTREADMILL RACE -fa "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("REFERENCE.FASTA"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -bam "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("MD.NM.TAGGED.BAM"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -bed "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("TARGETS.BED"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" --motif "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("KNONWN.MOTIF1 "),a("span",{pre:!0,attrs:{class:"token punctuation"}},[e._v("..")]),e._v("."),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -o "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("OUTPUT.RACE.DBSCAN.BIN"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v("\n")])])]),a("p",[e._v("RACE serializes the clustered repeats to a binary object from "),a("a",{attrs:{href:"https://docs.python.org/3/library/pickle.html",target:"_blank",rel:"noopener noreferrer"}},[e._v("pickle"),a("OutboundLink")],1),e._v(".\nIn addition to DBSCAN, RACE provides users with the possibility to group reads with similar repeat content using "),a("a",{attrs:{href:"https://en.wikipedia.org/wiki/Hierarchical_clustering",target:"_blank",rel:"noopener noreferrer"}},[e._v("Agglomerative Hierarchical Clustering"),a("OutboundLink")],1),e._v(". The first step is to create a full-dendogram tree of the clustered reads.")]),e._v(" "),a("div",{staticClass:"language-bash extra-class"},[a("pre",{pre:!0,attrs:{class:"language-bash"}},[a("code",[a("span",{pre:!0,attrs:{class:"token comment"}},[e._v("#with agglomerative hierarchical clustering")]),e._v("\nTREADMILL RACE -fa "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("REFERENCE.FASTA"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -bam "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("MD.NM.TAGGED.BAM"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -bed "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("TARGETS.BED"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" --motif "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("KNONW.MOTIF1 "),a("span",{pre:!0,attrs:{class:"token punctuation"}},[e._v("..")]),e._v("."),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -o "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("OUTPUT.RACE.DENDOGRAM.BIN"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" --hierarchical_clustering --dendogram\n")])])]),a("p",[e._v("Using this combination of flags, RACE serrializes the computed full-dendogram to a binary object from pickle and the dissimilarity array to a binary object from "),a("a",{attrs:{href:"https://numpy.org",target:"_blank",rel:"noopener noreferrer"}},[e._v("numpy"),a("OutboundLink")],1),e._v(". The computed full-dendogram can then be plotted using the dedicated "),a("a",{attrs:{href:"https://github.com/davidebolo1993/TREADMILL/blob/master/scripts/clusteranalysis.py",target:"_blank",rel:"noopener noreferrer"}},[e._v("clusteranalysis python script"),a("OutboundLink")],1),e._v(".")]),e._v(" "),a("div",{staticClass:"language-bash extra-class"},[a("pre",{pre:!0,attrs:{class:"language-bash"}},[a("code",[a("span",{pre:!0,attrs:{class:"token comment"}},[e._v("#python TREADMILL/scripts/clusteranalysis.py -h")]),e._v("\n\npython TREADMILL/scripts/clusteranalysis.py -d "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("OUTPUT.RACE.DENDOGRAM.BIN"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -s "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("OUTPUT.RACE.SIMMATRIX.BIN"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -o "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("OUTPUT.SILHOUETTE.FOLDER"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v("\n")])])]),a("p",[e._v("This script also performs "),a("a",{attrs:{href:"https://towardsdatascience.com/silhouette-method-better-than-elbow-method-to-find-optimal-clusters-378d62ff6891",target:"_blank",rel:"noopener noreferrer"}},[e._v("silhouette analysis"),a("OutboundLink")],1),e._v(" to identify the optimal number of clusters. Users can then choose the appropriate number of clusters RACE should output either by speficying the --clusters parameter (the number of clusters that best fits the dendogram or has higher silhouette score) or by specifying the --threshold parameter (the dissimilarity index where to cut the dendogram tree) in a new RACE run.")]),e._v(" "),a("div",{staticClass:"language-bash extra-class"},[a("pre",{pre:!0,attrs:{class:"language-bash"}},[a("code",[a("span",{pre:!0,attrs:{class:"token comment"}},[e._v("#specifying the number of clusters")]),e._v("\nTREADMILL RACE -fa "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("REFERENCE.FASTA"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -bam "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("MD.NM.TAGGED.BAM"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -bed "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("TARGETS.BED"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" --motif "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("KNONW.MOTIF1 "),a("span",{pre:!0,attrs:{class:"token punctuation"}},[e._v("..")]),e._v("."),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -o "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("OUTPUT.RACE.AGGLOMERATIVE.BIN"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" --hierarchical_clustering --clusters "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("CLUSTERS.INT"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v("\n\n"),a("span",{pre:!0,attrs:{class:"token comment"}},[e._v("#specifying a treshold to cut dendogram")]),e._v("\nTREADMILL RACE -fa "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("REFERENCE.FASTA"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -bam "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("MD.NM.TAGGED.BAM"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -bed "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("TARGETS.BED"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" --motif "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("KNONW.MOTIF1 "),a("span",{pre:!0,attrs:{class:"token punctuation"}},[e._v("..")]),e._v("."),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -o "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("OUTPUT.RACE.AGGLOMERATIVE.BIN"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" --hierarchical_clustering --threshold "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("TRESHOLD.FLOAT"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v("\n")])])]),a("p",[e._v("The flags accepted by TREADMILL RACE for fine-tuning of the decoy-alignment strategy as well as the clustering algorithms are summarized below:")]),e._v(" "),a("ol",[a("li",[e._v("The --affinity flag controls the eps parameter of the DBSCAN algorithm and is set to 70 by default (eps is calculated as the complement to 100 of the affinity value - that is, 30 by default)")]),e._v(" "),a("li",[e._v("The --support flag controls the minPts parameter of the DBSCAN algorithm as well as the minimum size that one cluster must have to be included in the final output. This is set to 5, by default")]),e._v(" "),a("li",[e._v("The --hierarchical_clustering flag tells RACE to perform Agglomerative Hierarchical Clustering instead of using DBSCAN. It must be used in combination with one between --dendogram, --threshold or --clusters")]),e._v(" "),a("li",[e._v("The --dendogram flag stores the full-dendogram resulting from Agglomerative Hierarchical Clustering to file, together with the pre-computed (dis-)similarity matrix used for clustering. These files are used for the silhouette analysis.")]),e._v(" "),a("li",[e._v("The --threshold flag allows users to cut the dendogram at a given treshold (dissimilarity index). This is usualy chosen upon manual inspection of the dendogram tree")]),e._v(" "),a("li",[e._v("The --cluster flag allows users to cut the dendogram so that exactly the chosen number of clusters is returned. The optimal number of clusters can be chosen upon manual inspection of the dendogram or by performing silhouette analysis")]),e._v(" "),a("li",[e._v("The --maxsize flag controls the maximum (approximate) number of repeats in the decoy reference used by RACE for the decoy-alignment strategy. This is set to 500 by default")]),e._v(" "),a("li",[e._v("The --flanking flag controls the number of flanking reference bases to include in each decoy chromosome. By default, each repeated segment is left- and right-flanked by 1000 reference bases")]),e._v(" "),a("li",[e._v("The --similarity flag controls the similarity percentage between consecutive decoy chromosomes. By default is set to 85")]),e._v(" "),a("li",[e._v("The --threads flag controls the number of threads used for the re-alignment step")]),e._v(" "),a("li",[e._v("The --store flag allows users to store in the output folder the decoy reference used by RACE together with the resulting reads-to-decoy alignment")]),e._v(" "),a("li",[e._v("The --plot flag allows users to store DBSCAN-clustering diagnostic plots (one for each locus in BED)")])]),e._v(" "),a("p",[e._v("Use cases for RACE are described in the "),a("RouterLink",{attrs:{to:"/usecases/usecases.html"}},[e._v("Use cases section")]),e._v(".")],1),e._v(" "),a("h2",{attrs:{id:"treadmill-trap"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#treadmill-trap"}},[e._v("#")]),e._v(" TREADMILL TRAP")]),e._v(" "),a("p",[e._v("TRAP profiles (approximate counting and genotyping) the clusters of repeats identified by RACE. First, TRAP computes a consensus sequence for each cluster, which drastically reduces the original error rate of the raw reads. It then performs approximate counting of repeats and genotype-likelihoods estimation.")]),e._v(" "),a("div",{staticClass:"language-bash extra-class"},[a("pre",{pre:!0,attrs:{class:"language-bash"}},[a("code",[a("span",{pre:!0,attrs:{class:"token comment"}},[e._v("#TREADMILL TRAP --help")]),e._v("\n\nTREADMILL TRAP -i "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("OUTPUT.RACE.DBSCAN/AGGLOMERATIVE.BIN"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -o "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("OUTPUT.RACE.FOLDER"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v("\n")])])]),a("p",[e._v("TRAP stores variant repetitions in (bgzipped) "),a("a",{attrs:{href:"https://samtools.github.io/hts-specs/VCFv4.3.pdf",target:"_blank",rel:"noopener noreferrer"}},[e._v("VCF format"),a("OutboundLink")],1),e._v(". INFO field of the VCF file contains custom informations:")]),e._v(" "),a("ol",[a("li",[e._v("END. The END coordinate of the repeat")]),e._v(" "),a("li",[e._v("MOTIF. The motif of the repeat, as specified by the user when running RACE")]),e._v(" "),a("li",[e._v("RN. The number of repeats in the reference sequence")]),e._v(" "),a("li",[e._v("RI. The motif and the number of repeat interruptions in the reference sequence")]),e._v(" "),a("li",[e._v("RALS. If present, the (consensus) sequence of the reference allele")]),e._v(" "),a("li",[e._v("RALD. If a reference allele is present, the edit distance-based similarity-score between reference and the allele")]),e._v(" "),a("li",[e._v("RALN. If a reference allele is present, the number of repeats in the reference allele")]),e._v(" "),a("li",[e._v("RALI. If a reference allele is present, the motif and the number of repeat interruptions in the reference allele")]),e._v(" "),a("li",[e._v("ALTD. If a reference allele is not present, the highest edit distance-based similarity score between the alternative allele/s and the reference sequence")]),e._v(" "),a("li",[e._v("AL1N/AL2N. If one ore more alternative alles are present, the number of repeats in the 1st/2nd alternative alleles")]),e._v(" "),a("li",[e._v("AL1I/AL2I. If one ore more alternative alles are present, the motif and the number of repeat interruptions in the 1st/2nd alternative alleles")]),e._v(" "),a("li",[e._v("SUBN. If --subgroups is specified and one or more alleles (other than those included in genotype) are present, the number of repeats in alleles that cluster alternatively")]),e._v(" "),a("li",[e._v("SUBP. If --subgroups is specified and one or more alleles (other than those included in genotype) are present, the relative abundance (percentage) of alleles that cluster alternatively")])]),e._v(" "),a("p",[e._v("Also, TRAP stores a (gzipped) tab-separated values file containing the original read names clustered by allele. This can be in turn used to call allele-specific methylation at each target locus.")]),e._v(" "),a("p",[e._v("A few more details on the parameters accepted by TREADMILL TRAP:")]),e._v(" "),a("ol",[a("li",[e._v("The --match/--mismatch/--gapopen/--gapextend flags control the the affine-gap penalties parameters for partial order alignment and consensus computation. By default, they are set to +5, -4, -8 and -6 respectively")]),e._v(" "),a("li",[e._v("The --reference_weight flag controls the percentage of reference sequence to add during consensus computation. By default, no reference sequence is added")]),e._v(" "),a("li",[e._v("The --substitution/--deletion/--insertion/--maxedit flags control the weighted, approximate counting of repeated motifs. If an approximate match has edit distance lower or equal than the value specified in --maxedit, then it is considered an approximate repetition and output in the RALI/AL1I/AL2I fields of the final VCF file (it is ignored otherwise). This is set to 1 by default")]),e._v(" "),a("li",[e._v("The --similarity flag determines the similarity percentage above which one allele is considered a reference allele. This is set to 80 by default")]),e._v(" "),a("li",[e._v("The --subgroups flag tells TRAP whether to output the relative abundance (SUBP field of the final VCF file) and the repeat content (SUBN field of the final VCF file) of minor alleles (that is, alleles that are not included in the main genotype)")])]),e._v(" "),a("p",[e._v("Use cases for TRAP are described in the "),a("RouterLink",{attrs:{to:"/usecases/usecases.html"}},[e._v("Use cases section")]),e._v(".")],1),e._v(" "),a("h2",{attrs:{id:"allele-specific-methylation-profiling"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#allele-specific-methylation-profiling"}},[e._v("#")]),e._v(" Allele-specific methylation profiling")]),e._v(" "),a("p",[e._v("TREADMILL integrates "),a("a",{attrs:{href:"https://github.com/bioinfomaticsCSU/deepsignal",target:"_blank",rel:"noopener noreferrer"}},[e._v("DeepSignal"),a("OutboundLink")],1),e._v(" to detect the methylation status of CG motifs. The dedicated "),a("a",{attrs:{href:"https://github.com/davidebolo1993/TREADMILL/blob/master/scripts/callmethylation.sh",target:"_blank",rel:"noopener noreferrer"}},[e._v("callmethylation bash script"),a("OutboundLink")],1),e._v(", provided in the scripts folder of TREADMILL, takes care of all the steps required to compute the allele-specific methylation frequencies at each target locus.")]),e._v(" "),a("div",{staticClass:"language-bash extra-class"},[a("pre",{pre:!0,attrs:{class:"language-bash"}},[a("code",[a("span",{pre:!0,attrs:{class:"token comment"}},[e._v("#bash TREADMILL/scripts/callmethylation.sh -h")]),e._v("\n\n"),a("span",{pre:!0,attrs:{class:"token function"}},[e._v("bash")]),e._v(" TREADMILL/scripts/callmethylation.sh -d "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("FAST5.FOLDER"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -F "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("TREADMILL.DECOY.FA"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -T "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("TREADMILL.ALLELES.TSV.GZ"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -o "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("OUTPUT.METHTYLATION.FOLDER"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -t "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("THREADS.NUMBER"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -g "),a("span",{pre:!0,attrs:{class:"token comment"}},[e._v("#do not use -g if running in cpu-mode ")]),e._v("\n")])])]),a("p",[e._v("The generated tab-separated values file with computed methylation frequencies can be in turn given to the companion "),a("a",{attrs:{href:"https://github.com/davidebolo1993/TREADMILL/blob/master/scripts/methanalysis.R",target:"_blank",rel:"noopener noreferrer"}},[e._v("methanalysis R script"),a("OutboundLink")],1),e._v(", along with original targets in BED. This script performs "),a("a",{attrs:{href:"https://en.wikipedia.org/wiki/Change_detection",target:"_blank",rel:"noopener noreferrer"}},[e._v("changepoint analysis"),a("OutboundLink")],1),e._v(" and identify regions that are differentially methylated along each allele.")]),e._v(" "),a("div",{staticClass:"language-bash extra-class"},[a("pre",{pre:!0,attrs:{class:"language-bash"}},[a("code",[a("span",{pre:!0,attrs:{class:"token comment"}},[e._v("#Rscript TREADMILL/scripts/methanalysis.R -h")]),e._v("\n\nRscript TREADMILL/scripts/methanalysis.R -f "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("METH.FREQUENCIES.TSV"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -o "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("OUTPUT.METHYLATION.FOLDER"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v(" -b "),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v("<")]),e._v("TARGETS.BED"),a("span",{pre:!0,attrs:{class:"token operator"}},[e._v(">")]),e._v("\n")])])]),a("p",[e._v("Use cases for these scripts are described in the "),a("RouterLink",{attrs:{to:"/usecases/usecases.html"}},[e._v("Use cases section")]),e._v(".")],1)])}),[],!1,null,null,null);t.default=r.exports}}]);