<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>General usage | TREADMILL documentation</title>
    <meta name="generator" content="VuePress 1.8.2">
    
    <meta name="description" content="">
    
    <link rel="preload" href="/treadmilldoc/assets/css/0.styles.b9c2e5d0.css" as="style"><link rel="preload" href="/treadmilldoc/assets/js/app.f13192c6.js" as="script"><link rel="preload" href="/treadmilldoc/assets/js/2.e9dff705.js" as="script"><link rel="preload" href="/treadmilldoc/assets/js/9.b9f07bb4.js" as="script"><link rel="prefetch" href="/treadmilldoc/assets/js/10.a629e91b.js"><link rel="prefetch" href="/treadmilldoc/assets/js/3.bd48039e.js"><link rel="prefetch" href="/treadmilldoc/assets/js/4.988f4cc4.js"><link rel="prefetch" href="/treadmilldoc/assets/js/5.9ae22b58.js"><link rel="prefetch" href="/treadmilldoc/assets/js/6.67696df0.js"><link rel="prefetch" href="/treadmilldoc/assets/js/7.11a62a51.js"><link rel="prefetch" href="/treadmilldoc/assets/js/8.1bcebbb0.js">
    <link rel="stylesheet" href="/treadmilldoc/assets/css/0.styles.b9c2e5d0.css">
  </head>
  <body>
    <div id="app" data-server-rendered="true"><div class="theme-container"><header class="navbar"><div class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/treadmilldoc/" class="home-link router-link-active"><!----> <span class="site-name">TREADMILL documentation</span></a> <div class="links"><div class="search-box"><input aria-label="Search" autocomplete="off" spellcheck="false" value=""> <!----></div> <nav class="nav-links can-hide"><div class="nav-item"><a href="/treadmilldoc/" class="nav-link">
  Home
</a></div><div class="nav-item"><a href="/treadmilldoc/installation/installation.html" class="nav-link">
  Installation
</a></div><div class="nav-item"><a href="/treadmilldoc/usage/usage.html" aria-current="page" class="nav-link router-link-exact-active router-link-active">
  General usage
</a></div><div class="nav-item"><a href="/treadmilldoc/usecases/usecases.html" class="nav-link">
  Use cases
</a></div><div class="nav-item"><a href="https://github.com/davidebolo1993/TREADMILL.git" target="_blank" rel="noopener noreferrer" class="nav-link external">
  GitHub
  <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></div> <!----></nav></div></header> <div class="sidebar-mask"></div> <aside class="sidebar"><nav class="nav-links"><div class="nav-item"><a href="/treadmilldoc/" class="nav-link">
  Home
</a></div><div class="nav-item"><a href="/treadmilldoc/installation/installation.html" class="nav-link">
  Installation
</a></div><div class="nav-item"><a href="/treadmilldoc/usage/usage.html" aria-current="page" class="nav-link router-link-exact-active router-link-active">
  General usage
</a></div><div class="nav-item"><a href="/treadmilldoc/usecases/usecases.html" class="nav-link">
  Use cases
</a></div><div class="nav-item"><a href="https://github.com/davidebolo1993/TREADMILL.git" target="_blank" rel="noopener noreferrer" class="nav-link external">
  GitHub
  <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></div> <!----></nav>  <ul class="sidebar-links"><li><a href="/treadmilldoc/installation/installation.html" class="sidebar-link">Installation</a></li><li><a href="/treadmilldoc/usage/usage.html" aria-current="page" class="active sidebar-link">General usage</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/treadmilldoc/usage/usage.html#treadmill-basic" class="sidebar-link">TREADMILL BASIC</a></li><li class="sidebar-sub-header"><a href="/treadmilldoc/usage/usage.html#treadmill-race" class="sidebar-link">TREADMILL RACE</a></li><li class="sidebar-sub-header"><a href="/treadmilldoc/usage/usage.html#treadmill-trap" class="sidebar-link">TREADMILL TRAP</a></li><li class="sidebar-sub-header"><a href="/treadmilldoc/usage/usage.html#allele-specific-methylation-profiling" class="sidebar-link">Allele-specific methylation profiling</a></li></ul></li><li><a href="/treadmilldoc/usecases/usecases.html" class="sidebar-link">Use cases</a></li></ul> </aside> <main class="page"> <div class="theme-default-content content__default"><h1 id="general-usage"><a href="#general-usage" class="header-anchor">#</a> General usage</h1> <p><a href="https://github.com/davidebolo1993/TREADMILL" target="_blank" rel="noopener noreferrer">TREADMILL<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a> is built on 3 modules:</p> <ol><li><a href="https://github.com/davidebolo1993/TREADMILL/tree/master/TREADMILL/BASIC" target="_blank" rel="noopener noreferrer">BASIC<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>, the 'BAm StatIstiCs' module</li> <li><a href="https://github.com/davidebolo1993/TREADMILL/tree/master/TREADMILL/RACE" target="_blank" rel="noopener noreferrer">RACE<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>, the 'ReAds ClusterEr' module</li> <li><a href="https://github.com/davidebolo1993/TREADMILL/tree/master/TREADMILL/TRAP" target="_blank" rel="noopener noreferrer">TRAP<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>, the 'Tandem RepeAts Profiler' module</li></ol> <p>It then wraps <a href="https://github.com/bioinfomaticsCSU/deepsignal" target="_blank" rel="noopener noreferrer">DeepSignal<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a> to perform allele-specific methylation profiling of CG motifs</p> <h2 id="treadmill-basic"><a href="#treadmill-basic" class="header-anchor">#</a> TREADMILL BASIC</h2> <p>BASIC allows users to retrieve essential statistics from targeted ONT sequencing runs in <a href="https://samtools.github.io/hts-specs/SAMv1.pdf" target="_blank" rel="noopener noreferrer">BAM format<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a> once a proper <a href="https://www.ensembl.org/info/website/upload/bed.html" target="_blank" rel="noopener noreferrer">BED file<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a> specifying repeat coordinates in enriced targets is provided. The input BAM file must be sorted and MD/NM-tagged.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#TREADMILL BASIC --help</span>

TREADMILL BASIC -bam <span class="token operator">&lt;</span>MD.NM.TAGGED.BAM<span class="token operator">&gt;</span> -bed <span class="token operator">&lt;</span>TARGETS.BED<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>OUTPUTBASIC.JSON.GZ<span class="token operator">&gt;</span>
</code></pre></div><p>BASIC dumps computed statistics to a (gzipped) <a href="https://www.json.org/json-en.html" target="_blank" rel="noopener noreferrer">JSON file<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>. This JSON file contains:</p> <ol><li>The number of unmapped, primary, secondary and supplementary reads in the BAM file</li> <li>The number of on-target and off-target primary alignments</li> <li>For on-target reads, the number of matches, mismatches, insertions, deletions and soft-clipped bases</li> <li>The lenght, quality and percent identity of on-target reads together with their per-base coverage</li></ol> <p>These informations can be summarized, for one ore more JSON file, using the companion <a href="https://github.com/davidebolo1993/TREADMILL/blob/master/scripts/plotstats.R" target="_blank" rel="noopener noreferrer">plotstats R script<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a> provided in the <a href="https://github.com/davidebolo1993/TREADMILL/tree/master/scripts" target="_blank" rel="noopener noreferrer">scripts folder of TREADMILL<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#Rscript TREADMILL/scripts/plotstats.R -h</span>

Rscript TREADMILL/scripts/plotstats.R -j <span class="token operator">&lt;</span>BASIC.JSON1.GZ,<span class="token punctuation">..</span>.<span class="token operator">&gt;</span> -l <span class="token operator">&lt;</span>EXP1,<span class="token punctuation">..</span>.<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>BASIC.STATS.PDF<span class="token operator">&gt;</span>
</code></pre></div><p>Use cases for BASIC are described in the <a href="/treadmilldoc/usecases/usecases.html">Use cases section</a>.</p> <h2 id="treadmill-race"><a href="#treadmill-race" class="header-anchor">#</a> TREADMILL RACE</h2> <p>RACE is the core module of TREADMILL. For each target in BED, TREADMILL RACE first applies a decoy-aligment strategy, following the principles described in <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-018-1505-2" target="_blank" rel="noopener noreferrer">STRetch<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>. It then runs a clustering algorithm - by default, the <a href="https://en.wikipedia.org/wiki/DBSCAN" target="_blank" rel="noopener noreferrer">Density-Based Spatial Clustering of Applications with Noise<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a> - to group reads with similar repeat content.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#TREADMILL RACE --help</span>

TREADMILL RACE -fa <span class="token operator">&lt;</span>REFERENCE.FASTA<span class="token operator">&gt;</span> -bam <span class="token operator">&lt;</span>MD.NM.TAGGED.BAM<span class="token operator">&gt;</span> -bed <span class="token operator">&lt;</span>TARGETS.BED<span class="token operator">&gt;</span> --motif <span class="token operator">&lt;</span>KNONWN.MOTIF1 <span class="token punctuation">..</span>.<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>OUTPUT.RACE.DBSCAN.BIN<span class="token operator">&gt;</span>
</code></pre></div><p>RACE serializes the clustered repeats to a binary object from <a href="https://docs.python.org/3/library/pickle.html" target="_blank" rel="noopener noreferrer">pickle<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>.
In addition to DBSCAN, RACE provides users with the possibility to group reads with similar repeat content using <a href="https://en.wikipedia.org/wiki/Hierarchical_clustering" target="_blank" rel="noopener noreferrer">Agglomerative Hierarchical Clustering<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>. The first step is to create a full-dendogram tree of the clustered reads.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#with agglomerative hierarchical clustering</span>
TREADMILL RACE -fa <span class="token operator">&lt;</span>REFERENCE.FASTA<span class="token operator">&gt;</span> -bam <span class="token operator">&lt;</span>MD.NM.TAGGED.BAM<span class="token operator">&gt;</span> -bed <span class="token operator">&lt;</span>TARGETS.BED<span class="token operator">&gt;</span> --motif <span class="token operator">&lt;</span>KNONW.MOTIF1 <span class="token punctuation">..</span>.<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>OUTPUT.RACE.DENDOGRAM.BIN<span class="token operator">&gt;</span> --hierarchical_clustering --dendogram
</code></pre></div><p>Using this combination of flags, RACE serrializes the computed full-dendogram to a binary object from pickle and the dissimilarity array to a binary object from <a href="https://numpy.org" target="_blank" rel="noopener noreferrer">numpy<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>. The computed full-dendogram can then be plotted using the dedicated <a href="https://github.com/davidebolo1993/TREADMILL/blob/master/scripts/clusteranalysis.py" target="_blank" rel="noopener noreferrer">clusteranalysis python script<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#python TREADMILL/scripts/clusteranalysis.py -h</span>

python TREADMILL/scripts/clusteranalysis.py -d <span class="token operator">&lt;</span>OUTPUT.RACE.DENDOGRAM.BIN<span class="token operator">&gt;</span> -s <span class="token operator">&lt;</span>OUTPUT.RACE.SIMMATRIX.BIN<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>OUTPUT.SILHOUETTE.FOLDER<span class="token operator">&gt;</span>
</code></pre></div><p>This script also performs <a href="https://towardsdatascience.com/silhouette-method-better-than-elbow-method-to-find-optimal-clusters-378d62ff6891" target="_blank" rel="noopener noreferrer">silhouette analysis<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a> to identify the optimal number of clusters. Users can then choose the appropriate number of clusters RACE should output either by speficying the --clusters parameter (the number of clusters that best fits the dendogram or has higher silhouette score) or by specifying the --threshold parameter (the dissimilarity index where to cut the dendogram tree) in a new RACE run.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#specifying the number of clusters</span>
TREADMILL RACE -fa <span class="token operator">&lt;</span>REFERENCE.FASTA<span class="token operator">&gt;</span> -bam <span class="token operator">&lt;</span>MD.NM.TAGGED.BAM<span class="token operator">&gt;</span> -bed <span class="token operator">&lt;</span>TARGETS.BED<span class="token operator">&gt;</span> --motif <span class="token operator">&lt;</span>KNONW.MOTIF1 <span class="token punctuation">..</span>.<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>OUTPUT.RACE.AGGLOMERATIVE.BIN<span class="token operator">&gt;</span> --hierarchical_clustering --clusters <span class="token operator">&lt;</span>CLUSTERS.INT<span class="token operator">&gt;</span>

<span class="token comment">#specifying a treshold to cut dendogram</span>
TREADMILL RACE -fa <span class="token operator">&lt;</span>REFERENCE.FASTA<span class="token operator">&gt;</span> -bam <span class="token operator">&lt;</span>MD.NM.TAGGED.BAM<span class="token operator">&gt;</span> -bed <span class="token operator">&lt;</span>TARGETS.BED<span class="token operator">&gt;</span> --motif <span class="token operator">&lt;</span>KNONW.MOTIF1 <span class="token punctuation">..</span>.<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>OUTPUT.RACE.AGGLOMERATIVE.BIN<span class="token operator">&gt;</span> --hierarchical_clustering --threshold <span class="token operator">&lt;</span>TRESHOLD.FLOAT<span class="token operator">&gt;</span>
</code></pre></div><p>The flags accepted by TREADMILL RACE for fine-tuning of the decoy-alignment strategy as well as the clustering algorithms are summarized below:</p> <ol><li>The --affinity flag controls the eps parameter of the DBSCAN algorithm and is set to 70 by default (eps is calculated as the complement to 100 of the affinity value - that is, 30 by default)</li> <li>The --support flag controls the minPts parameter of the DBSCAN algorithm as well as the minimum size that one cluster must have to be included in the final output. This is set to 5, by default</li> <li>The --hierarchical_clustering flag tells RACE to perform Agglomerative Hierarchical Clustering instead of using DBSCAN. It must be used in combination with one between --dendogram, --threshold or --clusters</li> <li>The --dendogram flag stores the full-dendogram resulting from Agglomerative Hierarchical Clustering to file, together with the pre-computed (dis-)similarity matrix used for clustering. These files are used for the silhouette analysis.</li> <li>The --threshold flag allows users to cut the dendogram at a given treshold (dissimilarity index). This is usualy chosen upon manual inspection of the dendogram tree</li> <li>The --cluster flag allows users to cut the dendogram so that exactly the chosen number of clusters is returned. The optimal number of clusters can be chosen upon manual inspection of the dendogram or by performing silhouette analysis</li> <li>The --maxsize flag controls the maximum (approximate) number of repeats in the decoy reference used by RACE for the decoy-alignment strategy. This is set to 500 by default</li> <li>The --flanking flag controls the number of flanking reference bases to include in each decoy chromosome. By default, each repeated segment is left- and right-flanked by 1000 reference bases</li> <li>The --similarity flag controls the similarity percentage between consecutive decoy chromosomes. By default is set to 85</li> <li>The --threads flag controls the number of threads used for the re-alignment step</li> <li>The --store flag allows users to store in the output folder the decoy reference used by RACE together with the resulting reads-to-decoy alignment</li> <li>The --plot flag allows users to store DBSCAN-clustering diagnostic plots (one for each locus in BED)</li></ol> <p>Use cases for RACE are described in the <a href="/treadmilldoc/usecases/usecases.html">Use cases section</a>.</p> <h2 id="treadmill-trap"><a href="#treadmill-trap" class="header-anchor">#</a> TREADMILL TRAP</h2> <p>TRAP profiles (approximate counting and genotyping) the clusters of repeats identified by RACE. First, TRAP computes a consensus sequence for each cluster, which drastically reduces the original error rate of the raw reads. It then performs approximate counting of repeats and genotype-likelihoods estimation.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#TREADMILL TRAP --help</span>

TREADMILL TRAP -i <span class="token operator">&lt;</span>OUTPUT.RACE.DBSCAN/AGGLOMERATIVE.BIN<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>OUTPUT.RACE.FOLDER<span class="token operator">&gt;</span>
</code></pre></div><p>TRAP stores variant repetitions in (bgzipped) <a href="https://samtools.github.io/hts-specs/VCFv4.3.pdf" target="_blank" rel="noopener noreferrer">VCF format<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>. INFO field of the VCF file contains custom informations:</p> <ol><li>END. The END coordinate of the repeat</li> <li>MOTIF. The motif of the repeat, as specified by the user when running RACE</li> <li>RN. The number of repeats in the reference sequence</li> <li>RI. The motif and the number of repeat interruptions in the reference sequence</li> <li>RALS. If present, the (consensus) sequence of the reference allele</li> <li>RALD. If a reference allele is present, the edit distance-based similarity-score between reference and the allele</li> <li>RALN. If a reference allele is present, the number of repeats in the reference allele</li> <li>RALI. If a reference allele is present, the motif and the number of repeat interruptions in the reference allele</li> <li>ALTD. If a reference allele is not present, the highest edit distance-based similarity score between the alternative allele/s and the reference sequence</li> <li>AL1N/AL2N. If one ore more alternative alles are present, the number of repeats in the 1st/2nd alternative alleles</li> <li>AL1I/AL2I. If one ore more alternative alles are present, the motif and the number of repeat interruptions in the 1st/2nd alternative alleles</li> <li>SUBN. If --subgroups is specified and one or more alleles (other than those included in genotype) are present, the number of repeats in alleles that cluster alternatively</li> <li>SUBP. If --subgroups is specified and one or more alleles (other than those included in genotype) are present, the relative abundance (percentage) of alleles that cluster alternatively</li></ol> <p>Also, TRAP stores a (gzipped) tab-separated values file containing the original read names clustered by allele. This can be in turn used to call allele-specific methylation at each target locus.</p> <p>A few more details on the parameters accepted by TREADMILL TRAP:</p> <ol><li>The --match/--mismatch/--gapopen/--gapextend flags control the the affine-gap penalties parameters for partial order alignment and consensus computation. By default, they are set to +5, -4, -8 and -6 respectively</li> <li>The --reference_weight flag controls the percentage of reference sequence to add during consensus computation. By default, no reference sequence is added</li> <li>The --substitution/--deletion/--insertion/--maxedit flags control the weighted, approximate counting of repeated motifs. If an approximate match has edit distance lower or equal than the value specified in --maxedit, then it is considered an approximate repetition and output in the RALI/AL1I/AL2I fields of the final VCF file (it is ignored otherwise). This is set to 1 by default</li> <li>The --similarity flag determines the similarity percentage above which one allele is considered a reference allele. This is set to 80 by default</li> <li>The --subgroups flag tells TRAP whether to output the relative abundance (SUBP field of the final VCF file) and the repeat content (SUBN field of the final VCF file) of minor alleles (that is, alleles that are not included in the main genotype)</li></ol> <p>Use cases for TRAP are described in the <a href="/treadmilldoc/usecases/usecases.html">Use cases section</a>.</p> <h2 id="allele-specific-methylation-profiling"><a href="#allele-specific-methylation-profiling" class="header-anchor">#</a> Allele-specific methylation profiling</h2> <p>TREADMILL integrates <a href="https://github.com/bioinfomaticsCSU/deepsignal" target="_blank" rel="noopener noreferrer">DeepSignal<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a> to detect the methylation status of CG motifs. The dedicated <a href="https://github.com/davidebolo1993/TREADMILL/blob/master/scripts/callmethylation.sh" target="_blank" rel="noopener noreferrer">callmethylation bash script<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>, provided in the scripts folder of TREADMILL, takes care of all the steps required to compute the allele-specific methylation frequencies at each target locus.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#bash TREADMILL/scripts/callmethylation.sh -h</span>

<span class="token function">bash</span> TREADMILL/scripts/callmethylation.sh -d <span class="token operator">&lt;</span>FAST5.FOLDER<span class="token operator">&gt;</span> -F <span class="token operator">&lt;</span>TREADMILL.DECOY.FA<span class="token operator">&gt;</span> -T <span class="token operator">&lt;</span>TREADMILL.ALLELES.TSV.GZ<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>OUTPUT.METHTYLATION.FOLDER<span class="token operator">&gt;</span> -t <span class="token operator">&lt;</span>THREADS.NUMBER<span class="token operator">&gt;</span> -g <span class="token comment">#do not use -g if running in cpu-mode </span>
</code></pre></div><p>The generated tab-separated values file with computed methylation frequencies can be in turn given to the companion <a href="https://github.com/davidebolo1993/TREADMILL/blob/master/scripts/methanalysis.R" target="_blank" rel="noopener noreferrer">methanalysis R script<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>, along with original targets in BED. This script performs <a href="https://en.wikipedia.org/wiki/Change_detection" target="_blank" rel="noopener noreferrer">changepoint analysis<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a> and identify regions that are differentially methylated along each allele.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#Rscript TREADMILL/scripts/methanalysis.R -h</span>

Rscript TREADMILL/scripts/methanalysis.R -f <span class="token operator">&lt;</span>METH.FREQUENCIES.TSV<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>OUTPUT.METHYLATION.FOLDER<span class="token operator">&gt;</span> -b <span class="token operator">&lt;</span>TARGETS.BED<span class="token operator">&gt;</span>
</code></pre></div><p>Use cases for these scripts are described in the <a href="/treadmilldoc/usecases/usecases.html">Use cases section</a>.</p></div> <footer class="page-edit"><!----> <!----></footer> <div class="page-nav"><p class="inner"><span class="prev">
      ←
      <a href="/treadmilldoc/installation/installation.html" class="prev">
        Installation
      </a></span> <span class="next"><a href="/treadmilldoc/usecases/usecases.html">
        Use cases
      </a>
      →
    </span></p></div> </main></div><div class="global-ui"></div></div>
    <script src="/treadmilldoc/assets/js/app.f13192c6.js" defer></script><script src="/treadmilldoc/assets/js/2.e9dff705.js" defer></script><script src="/treadmilldoc/assets/js/9.b9f07bb4.js" defer></script>
  </body>
</html>
